### ISSUE TITLE
Integration fails on HA 2025.1.0

### ISSUE THREAD

**Issue by skgsergio** [DurgNomis-drol/ha_toyota#319]
**Describe the bug**
When trying to install the integration on HA 2025.1.0 beta it fails resolving dependencies (see log below).

**To Reproduce**

1. Install HA 2025.1.0 beta
2. Check log

**Additional context**
```
[homeassistant.util.package] Unable to install package mytoyota>=2.1.4,<3.0: × No solution found when resolving dependencies:
  ╰─▶ Because mytoyota>=2.1.4 depends on pydantic>=1.10.0,<2.0.0 and
      pydantic==2.10.4, we can conclude that mytoyota>=2.1.4 cannot be used.
      And because only the following versions of mytoyota are available:
          mytoyota<=2.1.4
          mytoyota==2.2.0
      and you require mytoyota>=2.1.4, we can conclude that your requirements
      are unsatisfiable.
```

**Comment by hoffeck** [comment]
I can confirm

**Comment by desosav** [comment]
This must be an issue with the mytoyota library.
I have already opened an issue there: https://github.com/DurgNomis-drol/mytoyota/issues/382

**Comment by jvelaaa** [comment]
Same issue with 2025.1.0 final version

**Comment by modern74** [comment]
Same issue with 2025.1.0 final version

**Comment by GSA1974** [comment]
Idem pour moi. Désolé j'ai ouvert un autre post.
On peut supprimer.

**Comment by tobi-github** [comment]
Should hopefully be fixed soon

PR: [#384 DurgNomis-drol/mytoyota](https://github.com/DurgNomis-drol/mytoyota/pull/384)

**Comment by khzd** [comment]
+1
Same after migrating to 2025.1

**Comment by pc-bastler** [comment]
I can confirm (with release version 2025.1.0).

**Comment by NickyPot** [comment]
same here after upgrading to 2025.1.0

**Comment by GreJ0** [comment]
Same issue here after upgrading to 2025.1.0

**Comment by HEAzem** [comment]
Same issue here after upgrading to 2025.1.0

**Comment by jepdelpuru** [comment]
issue here after upgrading to 2025.1.0

**Comment by joost11111** [comment]
Same problem here

**Comment by pawlikmichal** [comment]
I have the same problem after updating to version 2025.1.0.

**Comment by rrromka** [comment]
+1

**Comment by daha76** [comment]
+1

**Comment by GSA1974** [comment]
Pour ma part ne fonctionne pas mieux avec la version 2025.1.1 Toujours le même problème.
Avez-vous désinstaller l'intégration puis réinstallée ?

**Comment by williamwwlr** [comment]
Moi j'ai fait une réinstalle sa marche toujours pas version 2025.1.1

**Comment by mgiebels** [comment]
+1

**Comment by sergitgna** [comment]
+1

**Comment by RealFriesi** [comment]
+1

**Comment by pctelco** [comment]
Les gars, écrire en français ici n'est pas la meilleure façon de communiquer. L'anglais est la langue utilisée ici.

Folks, writing in French here is not polite. We communicate in English.

I am German and write in English and when I don't know how to express it, I use Google Translate to help me.

But to get back to the topic, nothing can get the Toyota integration to run under HA Core 2025.1.x.

Is there already a solution in sight?

**Comment by Houbsi** [comment]
You have to wait. See below quote from the answer a little bit higher: https://github.com/DurgNomis-drol/ha_toyota/issues/319#issuecomment-2571392990

> Should hopefully be fixed soon
> 
> PR: [#384 DurgNomis-drol/mytoyota](https://github.com/DurgNomis-drol/mytoyota/pull/384)

**Comment by JaVier-Lloret** [comment]
I have same problem to update from 2024.12.5 to 2025.1.2.

2025-01-13 12:23:32.219 ERROR (SyncWorker_4) [homeassistant.util.package] Unable to install package mytoyota>=2.1.4,<3.0: × No solution found when resolving dependencies:
  ╰─▶ Because mytoyota>=2.1.4 depends on pydantic>=1.10.0,<2.0.0 and
      pydantic==2.10.4, we can conclude that mytoyota>=2.1.4 cannot be used.
      And because only the following versions of mytoyota are available:
          mytoyota<=2.1.4
          mytoyota==2.2.0
      and you require mytoyota>=2.1.4, we can conclude that your requirements
      are unsatisfiable.

Usualy I test all updates in a littel H.A. and after in my production H.A..

I'l wait the solution.

**Comment by ajain-93** [comment]
@CM000n  @peluho
Is it possible to pin this issue? I notice there are multiple new issues being created due the breaking changes in the `pydantic` library updated by 2025.1?

**Comment by raudraido** [comment]
> > @CM000n @peluho Is it possible to pin this issue? I notice there are multiple new issues being created due the breaking changes in the `pydantic` library updated by 2025.1?
> 
> And maybe block commenting, a ton of me too and +1 doesn't help nor provide valuable input.
> 
> I've already added info in the issue.

OR.. hear me out.. Fix the issue!

**Comment by skgsergio** [comment]
> > > @CM000n @peluho Is it possible to pin this issue? I notice there are multiple new issues being created due the breaking changes in the `pydantic` library updated by 2025.1?
> > 
> > 
> > And maybe block commenting, a ton of me too and +1 doesn't help nor provide valuable input.
> > I've already added info in the issue.
> 
> OR.. hear me out.. Fix the issue.. faster!

Or... hear me out: fix it yourself, create your own fork and maintain it.

This is a project by people that have a live and work and do it for fun, they maintain it best effort.

I'm not a maintainer of the project but I can surely say that your behavior is not welcome here.

And I am trying to be politically correct since it is not my project, if it were I would have been more blunt.

**Comment by jee67** [comment]
If you feel comfortable editing configuration files, there's a temporary workaround.
[https://github.com/DurgNomis-drol/mytoyota/pull/386#issuecomment-2597955506](https://github.com/DurgNomis-drol/mytoyota/pull/386#issuecomment-2597955506)

**Comment by skaker1** [comment]
> If you feel comfortable editing configuration files (/homeassistant/custom_components/toyota/manifest.json), there's a temporary workaround. [DurgNomis-drol/mytoyota#386 (comment)](https://github.com/DurgNomis-drol/mytoyota/pull/386#issuecomment-2597955506)

Worked fine!

**Comment by valerix85** [comment]
there is already a fix in progress https://github.com/DurgNomis-drol/mytoyota/pull/386#issuecomment-2597955506

**Comment by raudraido** [comment]
> > > > [@CM000n](https://github.com/CM000n) [@peluho](https://github.com/peluho) Is it possible to pin this issue? I notice there are multiple new issues being created due the breaking changes in the `pydantic` library updated by 2025.1?
> > > 
> > > 
> > > And maybe block commenting, a ton of me too and +1 doesn't help nor provide valuable input.
> > > I've already added info in the issue.
> > 
> > 
> > OR.. hear me out.. Fix the issue.. faster!
> 
> Or... hear me out: fix it yourself, create your own fork and maintain it.
> 
> This is a project by people that have a life and work and do it for fun, they maintain it best effort.
> 
> I'm not a maintainer of the project but I can surely say that your behavior is not welcome here.
> 
> And I am trying to be politically correct since it is not my project, if it were I would have been more blunt.

The F.. quick patch for this little dependency issue is already weeks sitting in pr, and nothing. Second month is running and still no fix.
Go on, take your time and spread love. IMO this is ridiculous.
Be as blunt as you possible can, don't care.

**Comment by valerix85** [comment]
> > > > > [@CM000n](https://github.com/CM000n) [@peluho](https://github.com/peluho) Is it possible to pin this issue? I notice there are multiple new issues being created due the breaking changes in the `pydantic` library updated by 2025.1?
> > > > 
> > > > 
> > > > And maybe block commenting, a ton of me too and +1 doesn't help nor provide valuable input.
> > > > I've already added info in the issue.
> > > 
> > > 
> > > OR.. hear me out.. Fix the issue.. faster!
> > 
> > 
> > Or... hear me out: fix it yourself, create your own fork and maintain it.
> > This is a project by people that have a life and work and do it for fun, they maintain it best effort.
> > I'm not a maintainer of the project but I can surely say that your behavior is not welcome here.
> > And I am trying to be politically correct since it is not my project, if it were I would have been more blunt.
> 
> The F.. quick patch for this little dependency issue is already weeks sitting in pr, and nothing. Second month is running and still no fix. Go on, take your time and spread love. IMO this is ridiculous. Be as blunt as you possible can, don't care.

Come on stop fucking around and wasting people's time with this crap.
The community doesn't need junkies like you

**Comment by dusopn** [comment]
> 
> The F.. quick patch for this little dependency issue is already weeks sitting in pr, and nothing. Second month is running and still no fix. Go on, take your time and spread love. IMO this is ridiculous. Be as blunt as you possible can, don't care.

You are still very welcome to do all the work yourself. We will be grateful.

**Comment by skgsergio** [comment]
> The F.. quick patch for this little dependency issue is already weeks sitting in pr, and nothing. Second month is running and still no fix. Go on, take your time and spread love. IMO this is ridiculous. Be as blunt as you possible can, don't care.

There is a workaround that works and persist HA updates, in the very same PR you mention.

But I guess you are too retarded to use it until the integration is updated as well as you are unable to maintain a single line of code or to behave in a public forum.

**Comment by raudraido** [comment]
retarded junkie here now understands what kind of behavior this community needs :D 
thanks

**Comment by tempico** [comment]
TLDR;

Here is the [working workaround](https://github.com/DurgNomis-drol/mytoyota/pull/386#issuecomment-2602188806) that survives Home Assistant restart. 

Edit the `manifest.json` file inside the `custom_components/toyota` Home Assistant configuration directory, so the requirements part of the `manifest.json` file looks like:
```  
"requirements": [
    "mytoyota @ git+https://github.com/DurgNomis-drol/mytoyota.git@refs/pull/386/head",
    "arrow"
  ],
```
Save the file and restart Home Assistant.

Otherwise, please allow developers little more time to polish things out before release.
Thank you for your patience 🙏

**Comment by halldark** [comment]
> "requirements": [
>     "mytoyota @ git+https://github.com/DurgNomis-drol/mytoyota.git@refs/pull/386/head",
>     "arrow"
>   ],
above
works as well in 2025.2

Previously provided: "mytoyota @ git+https://github.com/deejay1/mytoyota.git@simpler_first", is not working anymore.

**Comment by desosav** [comment]
> > "requirements": [
> > "mytoyota @ git+https://github.com/DurgNomis-drol/mytoyota.git@refs/pull/386/head",
> > "arrow"
> > ],
> > above
> > works as well in 2025.2
> 
> Previously provided: "mytoyota @ git+https://github.com/deejay1/mytoyota.git@simpler_first", is not working anymore.

Try the new workaround available here:
https://github.com/DurgNomis-drol/mytoyota/pull/386#issuecomment-2639082574

**Comment by sylvainvh** [comment]
Hello,

Just for information, the pull request with the integration of pydantic v2 has been merged last week. We should be able to reuse this integration

**Comment by raudraido** [comment]
> Hello,
> 
> Just for information, the pull request with the integration of pydantic v2 has been merged last week. We should be able to reuse this integration

🥇 ⏲

**Comment by joost11111** [comment]
> Hello,
> 
> Just for information, the pull request with the integration of pydantic v2 has been merged last week. We should be able to reuse this integration

Do we have to remove the workaround in the script and use the original? Or do we wait for a new version?

**Comment by jee67** [comment]
I still use
`    "mytoyota @ git+https://github.com/DurgNomis-drol/mytoyota",
`
and with that the integration works

![Image](https://github.com/user-attachments/assets/b00306a7-e9d7-4e65-8d2b-31f9944f48a3)

**Comment by CM000n** [comment]
Closed due to discontinuation of the project. See: https://github.com/DurgNomis-drol/ha_toyota/issues/346
